python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -89,6 +93,16 @@ def start(self, options: ExecutionOptions):
         self._cls = ray.remote(**self._ray_remote_args)(_MapWorker)
         for _ in range(self._autoscaling_policy.min_workers):
             self._start_actor()
+        refs = self._actor_pool.get_pending_actor_refs()
+
+        # We synchronously wait for the initial number of actors to start. This avoids
+        # situations where the scheduler is unable to schedule downstream operators
+        # due to lack of available actors, causing an initial "pileup" of objects on
+        # upstream operators, leading to a spike in memory usage prior to steady state.
+        logger.get_logger().info(
+            f"{self._name}: Waiting for {len(refs)} pool actors to start..."
+        )
```

Comment:
we can remove the max_tasks_in_flight ramp up if we are waiting for min_actors to start before scheduling any tasks.

Reply:
Good point! Reverted it.